sklearn/model_selection/_split.py

```diff
@@ -2685,6 +2687,154 @@ def _build_repr(self):
     return "%s(%s)" % (class_name, _pprint(params, offset=len(class_name)))
 
 
+class GroupTimeSeriesSplit(_BaseKFold):
+    """Time Series cross-validator variant with non-overlapping groups.
+
+    Provides train/test indices to split time series data samples that are
+    observed at fixed time intervals according to a third-party provided group.
+    In each split, test indices must be higher than before, and thus shuffling
+    in cross validator is inappropriate.
+
+    This cross-validation object is a variation of :class:`KFold`. In the kth
```

Comment:
I don't think it helps the reader much to call this a variation of KFold. The only apparent similarity is that test sets are non-overlapping?

Reply:
Removed.